Modules/rat_facts.py

```diff
@@ -0,0 +1,187 @@
+"""
+rat_facts.py - Managing Facts since 2018
+
+Copyright (c) 2018 The Fuel Rat Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+
+This module is built on top of the Pydle system.
+
+"""
+from utils.ratlib import Singleton
+from Modules.database_manager import DatabaseManager
+from Modules.context import Context
+import re
+import pyodbc
+import datetime
+
+internal_facts = ("!go", "!assign", "!add")
+# any fact name in here will not be handled by the handler
+
+
+class Fact:
+    name = None
```

Comment:
Im fairly certain these should not be class attributes unless `Fact` is a singleton?